onmt/inputters/inputter.py

```diff
@@ -314,23 +319,27 @@ def build_vocab(train_dataset_files, fields, data_type, share_vocab,
             del dataset
             gc.collect()
 
-    for name, field in fields["tgt"]:
-        _build_field_vocab(field, counters[name])
-        logger.info(" * %s vocab size: %d." % (name, len(field.vocab)))
-    if data_type == 'text':
-        for name, field in fields["src"]:
-            _build_field_vocab(field, counters[name])
-            logger.info(" * %s vocab size: %d." % (name, len(field.vocab)))
-        if share_vocab:
-            # `tgt_vocab_size` is ignored when sharing vocabularies
-            logger.info(" * merging src and tgt vocab...")
-            src_field = fields['src'][0][1]
-            tgt_field = fields['tgt'][0][1]
-            _merge_field_vocabs(
-                src_field, tgt_field, vocab_size=src_vocab_size,
-                min_freq=src_words_min_frequency)
-            logger.info(" * merged vocab size: %d." % len(src_field.vocab))
-
+    assert len(fields["tgt"]) == 1
+    tgt_multifield = fields["tgt"][0][1]
+    for tgt_name, tgt_field in tgt_multifield:
+        _build_field_vocab(tgt_field, counters[tgt_name])
```

Comment:
Could you incorporate the changes in #1199 into this?

Reply:
Yes, I've done so.